test/specs/parsing/axes/AncestorAxis.tests.ts

```diff
@@ -28,6 +30,26 @@ describe('ancestor', () => {
 		], documentNode);
 		chai.assert.deepEqual(evaluateXPathToStrings('ancestor-or-self::*/position()', documentNode.documentElement.firstChild), ['1', '2']);
 	});
+
+	it('passes buckets for ancestor', () => {
+		jsonMlMapper.parse([
+			'parentElement',
+			['childElement']
+		], documentNode);
+
+		const parentNode = documentNode.firstChild;
+		const childNode = documentNode.firstChild.firstChild;
+
+		const testDomFacade: IDomFacade = {
+			getParentNode: (node, bucket) => {
+				chai.assert.notEqual(bucket, null, 'There must be a bucket passed!');
+				return getBucketsForNode(node.parentNode).includes(bucket) ? node.parentNode : null;
```

Comment:
Seems like a somewhat creative way to check the right buckets. Lets use `chai.assert.include(getBucketsForNode(node.parentNode), bucket);` and just return `null`.

Reply:
The bucket comes from the selector, that's why `getBucketsFromNode(node.parentNode)` might not be include the bucket. for example, when parentNode is document node.